File: oauth_service/oauth/src/main/java/com/oauth/oauth/model/Grade.java
package com.oauth.oauth.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Grade {
    private String courseId;
    private String courseName;
    private Double grade;
    private String semester;
}

File: grades_service/app/schemas/grade.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class GradeBase(BaseModel):
    student_id: str
    student_name: str
    course_id: str
    course_name: str
    grade: float = Field(..., ge=0, le=100)
    semester: str
    professor_id: str
    professor_name: str
    comments: Optional[str] = None

class GradeCreate(GradeBase):
    pass

class GradeUpdate(BaseModel):
    grade: Optional[float] = Field(None, ge=0, le=100)
    comments: Optional[str] = None

class GradeResponse(GradeBase):
    id: str
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True
